include/boost/process/v2/impl/exe.ipp

```diff
@@ -0,0 +1,313 @@
+
+#include <boost/process/v2/detail/config.hpp>
+#include <boost/process/v2/detail/last_error.hpp>
+#include <boost/process/v2/detail/throw_error.hpp>
+#include <boost/process/v2/detail/exe.hpp>
+#include <boost/process/v2/exe.hpp>
+
+#if defined(BOOST_PROCESS_V2_WINDOWS)
+#include <windows.h>
+#else
+#include <climits>
+
+#if (defined(__APPLE__) && defined(__MACH__))
+#include <libproc.h>
+#endif
+
+#if (defined(BOOST_PROCESS_V2_WINDOWS) || defined(__linux__) || defined(__ANDROID__) || defined(__sun))
+#include <cstdlib>
+#endif
+
+#if (defined(__FreeBSD__) || defined(__DragonFly__) || defined(__NetBSD__))
+#include <sys/types.h>
+#include <sys/sysctl.h>
+#endif
+
+#if defined(__OpenBSD__)
+#include <sys/types.h>
+#include <sys/param.h>
+#include <sys/sysctl.h>
+#include <kvm.h>
+#endif
+
+BOOST_PROCESS_V2_BEGIN_NAMESPACE
+
+namespace ext {
+
+#if defined(BOOST_PROCESS_V2_WINDOWS)
+
+boost::filesystem::path exe_path(pid_type pid, error_code & ec)
+{
+    boost::filesystem::path path;
+    if (pid == GetCurrentProcessId()) 
+    {
+        wchar_t buffer[MAX_PATH];
+        if (GetModuleFileNameW(nullptr, buffer, sizeof(buffer)) != 0) 
+        {
+            wchar_t exe[MAX_PATH];
+            if (_wfullpath(exe, buffer, MAX_PATH)) 
+            {
+                path = exe;
+            }
+        }
+    } 
+    else 
+    {
+        HANDLE proc = detail::ext::open_process_with_debug_privilege(pid);
+        if (proc == nullptr) return path;
+        wchar_t buffer[MAX_PATH];
+        DWORD size = sizeof(buffer);
+        if (QueryFullProcessImageNameW(proc, 0, buffer, &size) != 0) 
+        {
+            wchar_t exe[MAX_PATH];
+            if (_wfullpath(exe, buffer, MAX_PATH)) 
+            {
+                path = exe;
+            }
+        }
+        CloseHandle(proc);
+    }
+    if (path.string().empty())
+    {
+        ec = detail::get_last_error();
+    }
+    return path;
+}
+
+#elif (defined(__APPLE__) && defined(__MACH__))
+
+boost::filesystem::path exe_path(pid_type pid, error_code & ec)
+{
+    std::boost::filesystem::path;
+    char exe[PROC_PIDPATHINFO_MAXSIZE];
+    if (proc_pidpath(pid, exe, sizeof(exe)) > 0) 
+    {
+        char buffer[PATH_MAX];
+        if (realpath(exe, buffer)) 
+        {
+            path = buffer;
+        }
+    }
+    if (path.string().empty())
+    {
+        ec = detail::get_last_error();
+    }
+    return path;
+}
+
+#elif (defined(__linux__) || defined(__ANDROID__))
+
+boost::filesystem::path exe_path(pid_type pid, error_code & ec)
+{
+    std::boost::filesystem::path;
+    char exe[PATH_MAX];
+    if (realpath(("/proc/" + std::to_string(pid) + "/exe").c_str(), exe)) 
+    {
+        path = exe;
+    }
+    if (path.string().empty())
+    {
+        ec = detail::get_last_error();
+    }
+    return path;
+}
+
+#elif defined(__FreeBSD__) || defined(__DragonFly__)
+
+boost::filesystem::path exe_path(pid_type pid, error_code & ec) 
+{
+    std::boost::filesystem::path;
```

Comment:
can I get a few comments of what this is?

Reply:
that was a typo, isn't it just without the std::?